packages/core/src/components/Autocomplete/Autocomplete.jsx

```diff
@@ -0,0 +1,174 @@
+import Button from '../Button/Button';
+import Downshift from 'downshift';
+import PropTypes from 'prop-types';
+import React from 'react';
+import TextField from '../TextField/TextField';
+import uniqueId from 'lodash.uniqueid';
+
+/**
+ * Determine if a React component is a TextField
+ * @param {React.Node} child - a React component
+ * @return {Boolean} Is this a TextField component?
+ */
+function isTextField(child) {
+  return child != null && child.type === TextField;
+}
+
+/**
+ * The `Autocomplete` component is a parent component that adds autocomplete functionality to a `TextField` component.
+ */
+export class Autocomplete extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.id = uniqueId('autocomplete_');
+    this.labelId = uniqueId('constrained-list-header_');
+  }
+
+  renderChildren(getInputProps) {
+    // Extend props on the TextField, by passing them through
+    // Downshift's `getInputProps` method
+    return React.Children.map(this.props.children, child => {
+      if (isTextField(child)) {
+        return React.cloneElement(
+          child,
+          getInputProps({
+            fieldClassName: 'ds-c-autocomplete__input',
+            id: this.id,
+            labelClassName: 'ds-u-margin-top--0 ds-c-autocomplete__label'
+          })
+        );
+      }
+
+      return child;
+    });
+  }
+
+  render() {
+    const {
+      clearAriaLabel,
+      clearInputText,
+      items,
+      itemToString,
+      label,
+      onChange
+    } = this.props;
+
+    return (
+      <Downshift
+        itemToString={itemToString}
+        onChange={onChange}
+        render={({
+          clearSelection,
+          getInputProps,
+          getItemProps,
+          highlightedIndex,
+          isOpen,
+          inputValue
+        }) => (
+          <div className="ds-u-clearfix ds-c-autocomplete">
+            {this.renderChildren(getInputProps)}
+
+            {isOpen ? (
+              <div className="ds-u-border--1 ds-u-padding--1 ds-c-autocomplete__list">
+                {label && (
+                  <h5
+                    className="ds-u-margin--0 ds-u-padding--1 ds-u-padding-left--2"
+                    id={this.labelId}
+                  >
+                    {label}
+                  </h5>
+                )}
+
+                <ul
+                  aria-labelledby={label ? this.labelId : null}
+                  className="ds-c-list--bare"
+                  role="listbox"
+                >
+                  {items
+                    .filter(
```

Comment:
How would this work if you have to query an API..is there a loading state? 

@pwolfert Does this implementation work for how y'all are thinking about the zipcode + county picker? I assume you would be doing the filtering on the backend and only returning the counties that match?

Reply:
For our case the filtering would be unnecessary, yeah. We'd only be giving items to the component that we want it to render.

Reply:
Removed the filtering, per @pwolfert comment to it not being needed.